### PR TITLE
Race fix. Changes modelStructsMap implementation from map with mutex to sync.Map

### PR DESCRIPTION
**Pull request by mtfelian**
### What did this pull request do?

- _Changes modelStructsMap implementation from map with mutex to sync.Map_
- Fixes https://github.com/jinzhu/gorm/issues/1407

### CODE REVIEW

File: model_struct.go [hunk 1]
```diff
@@ -17,26 +17,7 @@ var DefaultTableNameHandler = func(db *DB, defaultTableName string) string {
...
-	defer s.l.RUnlock()
-	return s.m[key]
-}
-
-func newModelStructsMap() *safeModelStructsMap {
-	return &safeModelStructsMap{l: new(sync.RWMutex), m: make(map[reflect.Type]*ModelStruct)}
-}
+func newModelStructsMap() sync.Map { return sync.Map{} }
 
 var modelStructsMap = newModelStructsMap()
```
Comment (jinzhu) [inline model_struct.go]: Just replace this with `var modelStructsMap sync.Map`

Comment (mtfelian) [reply model_struct.go]: Done.

**Comment by mtfelian** [comment]
Some issue with wercker mariadb image seems not related to this change.

**Comment by mtfelian** [comment]
@jinzhu let's merge it? This issue breaks our tests with race detector. Replaced imports to my fork. Ugly.

**Comment by mtfelian** [comment]
Seems this doesn't fix the original issue as @dnathe4th reported. Posted #2060

**Pull request merged**